app/scripts/controllers/incoming-transactions.js

```diff
@@ -161,28 +165,26 @@ export default class IncomingTransactionsController {
     this.store.updateState({
       incomingTxLastFetchedBlocksByNetwork: {
         ...currentBlocksByNetwork,
-        [network]: newLatestBlockHashByNetwork,
+        [CHAIN_ID_TO_TYPE_MAP[chainId]]: newLatestBlockHashByNetwork,
```

Comment:
`CHAIN_ID_TO_TYPE_MAP[chainId]` will evaluate to `undefined` if the `chainId` isn't for any of the default networks. There are other instances of this that will cause the same problem. We should probably create a `getNetworkIdentifier(chainId: string)` utility function that returns e.g. `'rpc'` if the `chainId` is not in the map.

Reply:
Or, alternatively, start keying `incomingTxLastFetchedBlocksByNetwork` by `chainId` and add a migration.

Reply:
Ah, good catch! This is no good. We shouldn't really have been keeping the network-keyed data in state either, as it was always blank.

I think I'd rather ensure that we bail early if we're on a non-tracked network, and set nothing in state for that case.

Reply:
Oh wow, I completely missed that we only track the default networks. 🤦 

That being the case, I agree that we should bail!

Reply:
I have finally pushed this change. It took much longer than expected to properly unit test this.